### PR TITLE
Add source code for Chapter 9

### PR DESCRIPTION
**Pull request by Peter-VanWeert**

### CODE REVIEW

File: Exercises/NoModules/Chapter 09/Soln9_02.cpp [hunk 1]
```diff
@@ -0,0 +1,55 @@
+// Exercise 9-2 Working with std::string_view<> and std::span<>
+#include <iostream>
+#include <format>
+#include <string_view>
+#include <span>
+
+// The function prototype including defaults for parameters
+void show_data(
+  std::span<const int> data,
```
Comment (MarcGregoire) [inline Exercises/NoModules/Chapter 09/Soln9_02.cpp:9]: My text says:
"Exercise 9-2. Adjust Ex8_11 to use std::string_view"
but you are converting 8_11 to using string_view and span.
Did you update the exercise in the text?

File: Exercises/NoModules/Chapter 09/Soln9_06.cpp [hunk 2]
```diff
@@ -0,0 +1,23 @@
+// Exercise 9-5. Passing a vector to a fixed-size std::span<>
```
Comment (MarcGregoire) [inline Exercises/NoModules/Chapter 09/Soln9_06.cpp:1]: 9-6

**Review by MarcGregoire — CHANGES_REQUESTED** [review]

File: Examples/NoModules/Chapter 09/Ex9_01.cpp [hunk 3]
```diff
@@ -0,0 +1,46 @@
+// Working with std::optional<>
+#include <optional>     // std::optional<> is defined in the <optional> module
```
[the remark this answers is not in the document]
Comment (Peter-VanWeert) [reply Examples/NoModules/Chapter 09/Ex9_01.cpp:2]: True that. Fixed.

File: Exercises/NoModules/Chapter 09/Soln9_02.cpp (hunk 1 above)
Comment (Peter-VanWeert) [reply Exercises/NoModules/Chapter 09/Soln9_02.cpp:9]: I did just that: "Adjust Ex8_11 to use std::string_view and std::span<>."

**Pull request merged**